pdns/recursordist/rec-system-resolve.cc

```diff
@@ -304,3 +304,26 @@ void pdns::RecResolve::Refresher::trigger()
   wakeup = true;
   condVar.notify_one();
 }
+
+ComboAddress pdns::fromNameOrIP(const string& str, uint16_t defPort, Logr::log_t log)
+{
+  try {
+    ComboAddress addr = parseIPAndPort(str, defPort);
+    return addr;
+  }
+  catch (const PDNSException&) {
+    uint16_t port = defPort;
+    string::size_type pos = str.rfind(':');
+    if (pos != string::npos) {
+      port = pdns::checked_stoi<uint16_t>(str.substr(pos + 1));
+    }
+    auto& res = pdns::RecResolve::getInstance();
+    ComboAddress address = res.lookupAndRegister(str.substr(0, pos), time(nullptr));
```

Comment:
If I'm not mistaken this means that the resolve callback will be called when a name associated to a catalog or RPZ zone resolves to a different IP. Do we want to actually act on it? As far as I can tell this would not trigger a reload of the catalog/RPZ zones, as opposed to issuing a `rec_control reload-lua-config`.

Reply:
Indeed, if the IP of a named primary for RPZ or Catz changes, this will trigger a (likely unneeded) reload of the forwarding zones. I thought about this issue while building this and decided it isn't a very big deal, assuming that those do not change all the time.

In the future we might want to associate a list of callbacks to a specific name, so we can take the right action if the name resolves to a new value. A list because a a name can be used for multiple purposes: e.g. both as RPZ primary and as forwarding target. In that case we also might want to be more specific, e.g. only update forwarding zones that actually use that name. This all requires some careful design.

Reply:
Agreed, I'm not too worried about the unneeded reload. :+1: